order_analysis.rb

```diff
@@ -24,13 +27,56 @@ def extract_customer(line)
       return
     end
 
-    customer_id = parts[0]
-    first_name = parts[1]
-    last_name = parts[2]
-    location = parts[3]
+    customer = Customer.new(parts[0], parts[1], parts[2], parts[3])
+    @customers.push(customer)
 
-    puts "Customer #{customer_id}: #{first_name} #{last_name} (#{location})"
+  end
+
+  def analyze_orders
+    orders_csv = File.read("orders.csv")
+    orders_lines = orders_csv.split("\n")
+
+    orders_lines.each do |order_line|
+      order = extract_order(order_line)
+      @customers.each do |customer|
+        customer.order_to_customer(order)
+      end
+    end
+
+  end
+  def extract_order(line)
+    parts = line.split(CSV_DELIMITER)
+    if parts.size != 3
+      puts "[ERROR] Something's wrong with this order line: #{line}!"
+      return
+    end
+    order = Order.new(parts[0], parts[1], parts[2])
+  end
+  def output
+    @customers.each do |customer|
```

Comment:
You could write this in the shortened version I think. I believe, this should work, but try it first (you have to delete all lines of the block:

```suggestion
    @customers.each(&:output)
```